Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -3830,6 +3829,35 @@ void MediaPlayerPrivateGStreamer::flushCurrentBuffer()
 }
 #endif
 
+void MediaPlayerPrivateGStreamer::setVisibleInViewport(bool isVisible)
+{
+    if (isMediaStreamPlayer())
+        return;
+
+    // Some layout tests (webgl) expect playback of invisible videos to not be suspended, so allow
+    // this using an environment variable, set from the webkitpy glib port sub-classes.
+    const char* allowPlaybackOfInvisibleVideos = g_getenv("WEBKIT_GST_ALLOW_PLAYBACK_OF_INVISIBLE_VIDEOS");
+    if (!isVisible && allowPlaybackOfInvisibleVideos && !strcmp(allowPlaybackOfInvisibleVideos, "1"))
+        return;
+
+    GST_INFO_OBJECT(m_pipeline.get(), "%s %s player %svisible in viewport", m_isMuted ? "Muted" : "Un-muted", m_player->isVideoPlayer() ? "video" : "audio", isVisible ? "" : "no longer ");
```

Comment:
`%svisible` ?

Reply:
yes?

Reply:
Reading the code of GStreamer's `printf_parse()` this is supported, parsing stops after the `s` was detected.